opm/output/eclipse/RestartIO.cpp

```diff
@@ -509,8 +541,10 @@ void save(const std::string& filename,
           data::Wells wells,
           const EclipseState& es,
           const EclipseGrid& grid,
+          std::map<std::string, std::vector<double>> extra_data,
```

Comment:
>The wells and extra_data arguments should be passed as const references 

The reason these are passed by value is to support asyncronous output, where calling scope can manipulate these variables immediately after the call is issued.